### ISSUE TITLE
CDN for S3 external storage

### ISSUE THREAD

**Issue by Kitanit** [nextcloud/server#30917]
Ability to specify CDN for external storage
*Example: when using CloudFlare with B2, the traffic will be free, and as now you have to address directly

**Comment by dw5** [comment]
I agree, it be nice being not limited to Vultr or cherry servers, also same thing hopefully for use as primary S3 storage https://docs.nextcloud.com/server/latest/admin_manual/configuration_files/primary_storage.html

Until then I guess rclone mount or serve https://rclone.org/commands/rclone_serve_webdav/ + https://rclone.org/b2/#b2-download-url or https://rclone.org/s3/#s3-download-url can be a dirty patch?
